--- ./src/config.js
+++ ./src/config.js
@@ -132,11 +132,50 @@ let prometheusConfig = new PrometheusConfig();
 let emailConfig = new EmailConfig();
 let captchaConfig = new CaptchaConfig();
 
+let configFileLocationsEnvironmentMap = {
+    "config.yaml": "CYTUBE_CONFIG_YAML_LOCATION"
+};
+
 /**
  * Initializes the configuration from the given YAML file
  */
 exports.load = function (file) {
-    let absPath = path.join(__dirname, "..", file);
+    let absPath = null;
+
+    console.log("HELLO WORLD");
+    console.log(file);
+    console.log(configFileLocationsEnvironmentMap[file]);
+
+    if (configFileLocationsEnvironmentMap[file] != null) {
+        let envVarName = configFileLocationsEnvironmentMap[file];
+
+        console.log("envVarName:");
+        console.log(envVarName);
+        console.log(process.env[envVarName]);
+
+        if (process.env[envVarName] != null) {
+            console.log("process.env[envVarName] is not null");
+            let fileLocation = process.env[envVarName];
+
+            console.log("fileLocation:");
+            console.log(fileLocation);
+
+            var fs = require("fs");
+
+            if (fs.existsSync(fileLocation)) {
+                console.log("fileLocation exists in filesystem!");
+                absPath = fileLocation;
+                console.log("absPath: ", absPath);
+            }
+        }
+
+        console.log("/HELLO WORLD");
+    }
+
+    if (absPath == null) {
+        absPath = path.join(__dirname, "..", file);
+    }
+
     try {
         cfg = YAML.load(absPath);
     } catch (e) {
